### DISCUSSION TITLE
Rust as an optional dependency

### DISCUSSION THREAD

**Discussion by LilyFirefly** [PyO3/maturin#1277] [Q&A] [answered]
I'm working on adding a rust extension to a python profiling tool to try and get a speed improvement. However, I'm relying on `pyo3::ffi::PyEval_SetProfile`, which is not available on PyPy.

Is there a recommended way to build wheels for CPython that include my rust extension (and my pure python code) and build wheels for PyPy that only include my pure python code?

**Comment by messense** [comment, marked as answer]
No support for making Rust extension optional, see https://github.com/PyO3/maturin/issues/1081.

Sounds like you want https://docs.rs/pyo3-build-config/latest/pyo3_build_config/, for example

* https://github.com/pydantic/pydantic-core/blob/e146483e2becb6151e76539861db0053465bb22a/Cargo.toml#L64-L67
* https://github.com/pydantic/pydantic-core/blob/e146483e2becb6151e76539861db0053465bb22a/build.rs#L27

Then you can use `#[cfg(not(PyPy))]` in your code.

**Reply by LilyFirefly** [reply]
Thanks! Once I put `build.rs` in the right place (not in `src`!) this worked great!
